### DISCUSSION TITLE
No network

### DISCUSSION THREAD

**Discussion by loopmasta** [utmapp/UTM#2661] [UTM Help (macOS)] [answered]
I can't get a network running with any of the UTM ARM64 debian 10.4 images listed at the utmApp gallery.
Am I missing something or is this a known bug? Any help is greatly appreciated. Thanks.

UTM: 2.1.1 (30)
MacOs: 11.4 (20F71)
Apple M1

**Comment by jamesmortensen** [comment]
I wasn't able to load any web pages, but this is what I tried:

```
$ ip -c link show
```

Above command listed out lo and enp0s6 as network interfaces.  I then tried configuring a static IP in /etc/network/interfaces:

```
auto enp0s6
iface enp0s6 inet static
address 10.x.x.x
netmask 255.255.255.0
gateway x.x.x.x
```

I removed the dhcp configuration as described here:  https://www.cyberciti.biz/faq/add-configure-set-up-static-ip-address-on-debianlinux/

I also added some DNS servers to /etc/resolv.conf:

```
nameserver 8.8.8.8
nameserver 1.1.1.1
```

I then restarted the networking service:

```
sudo systemctl restart networking.service
```

Then running `/sbin/ifconfig` showed that the interface enp0s6 IP was 10.x.x.x; however, I am still unable to connect to any web pages.  There doesn't seem to be any information on how to get the UTM Debian 10 VM internet working.  Ubuntu and ReactOS both connected to the internet out of the box for me.  Hoping this might help us figure out what the problem is.

I'm also wondering if enabling networking is something that doesn't involve configuring Debian and if we're supposed to configure this in the VM settings in the UTM interface.

**Comment by jamesmortensen** [comment]
Today, I booted up the Debian 10.4 ARM LXDE VM, and was able to load pages in the browser.  Below is a screenshot of the network settings in `/etc/network/interfaces`. I went back to using DHCP, but using the interface name that was showing when I ran `ip -c link show`.

<img width="1024" alt="Screenshot 2021-07-22 at 10 15 01 AM" src="https://user-images.githubusercontent.com/1315816/126591448-df72ee08-09a4-4172-b18f-8193c3b67fab.png">

Oddly, while starting a fresh Debian 10.4 ARM Xfce VM, networking worked out of the box.  Not sure if there is a difference in the configuration between the two Debian VM's, or if some setting was unwittingly changed in the UTM app to enable networking in both.

Hope this information is helpful.

**Comment by Damiloju** [comment, marked as answer]
If anyone comes here, I had the same issues with Ubuntu 20.04

>nameserver 8.8.8.8
nameserver 1.1.1.1

Added those to /etc/resolv.conf as mentioned and restarted network-manager.service

>sudo systemctl restart network-manager.service

things worked after that

**Reply by loopmasta** [reply]
@Damiloju This solution works most of the times but fails sometimes for no reason and I have to restart the network-manager.service. 

@jamesmortensen had a similar solution for /etc/resolv.conf but his additions in /etc/network/interfaces didn't work for me.

I mark this post as the answer because it kinda works for me with Ubuntu 20.04. Thanks a lot.

**Reply by rvgeerligs** [reply]
On the debian images still no network.
debian@debian:~$ sudo route -n
Kernel IP routing table
Destination Gateway Genmask Flags Metric Ref Use Iface
0.0.0.0 10.0.2.2 0.0.0.0 UG 0 0 0 enp0s6
10.0.2.0 0.0.0.0 255.255.255.0 U 0 0 0 enp0s6

However, there is no link to 0.0.0.0.
All changes I make to /etc/resolv.conf is overwritten in a few seconds with the value I entered in Networking>Advanced.

**Comment by rvgeerligs** [comment]
On the debian images still no network.
debian@debian:~$ sudo route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         10.0.2.2        0.0.0.0         UG    0      0        0 enp0s6
10.0.2.0        0.0.0.0         255.255.255.0   U     0      0        0 enp0s6

However, there is no link to 0.0.0.0.
All changes I make to /etc/resolv.conf is overwritten in a few seconds with the value I entered in Networking>Advanced.

**Comment by glingy** [comment]
Since I found this while looking for a fix for the same issue, what's listed at the bottom of the ubuntu gallery worked for me:
https://mac.getutm.app/gallery/ubuntu-20-04
```
When switching between display modes, the network adapter may be renamed by Ubuntu. To fix this, you need to set up networking again.

- Run ip link show and look at the last adapter name. For example, it may be listed as enp0s9.
- Edit /etc/netplan/00-installer-config.yaml and copy your configuration for enp0s8 (or whatever the old adapter was named) and paste it immediately after for enp0s9 (or whatever the new adapter is named).
- Reboot and you should be able to use networking in both display modes.
```

**Reply by samuelmarciano** [reply]
And what if you have only one adapter :network:
```
  ethernets:
    enp0s10:
      dhcp4: true
  version: 2
```
